linter/linter.go

```diff
@@ -606,6 +621,40 @@ func (l *Linter) lintSubRoutineDeclaration(decl *ast.SubroutineDeclaration, ctx
 	return types.NeverType
 }
 
+func (l *Linter) lintGotoStatement(stmt *ast.GotoStatement, ctx *context.Context) types.Type {
+	// validate destination name
+	if !isValidName(stmt.Destination.Value) {
+		l.Error(InvalidName(stmt.Destination.GetMeta(), stmt.Destination.Value, "goto").Match(GOTO_SYNTAX))
+	}
+
+	if err := ctx.AddGoto(stmt.Destination.Value, &types.Goto{Decl: stmt}); err != nil {
+		e := &LintError{
+			Severity: ERROR,
+			Token:    stmt.Destination.GetMeta().Token,
+			Message:  err.Error(),
+		}
+		l.Error(e.Match(GOTO_DUPLICATED))
+	}
+
+	return types.NeverType
+}
+
+func (l *Linter) lintGotoDestinationStatement(stmt *ast.GotoDestinationStatement, ctx *context.Context) types.Type {
+	if gd, ok := ctx.Gotos[stmt.Name.Value]; ok {
+		if gd.IsUsed {
+			l.Error(DuplicatedUseForGotoDestination(stmt.GetMeta(), stmt.Name.Value))
+			return types.NullType
+		}
+
+		gd.IsUsed = true
+		return types.GotoType
+	} else {
+		l.Error(UndefinedGotoDestination(stmt.GetMeta(), stmt.Name.Value))
+	}
```

Comment:
(If I misunderstood, let me know)
The `goto` statement should have a scope conception, otherwise,

```
sub some_function {
  goto foo;
}

sub another_function {
  foo:
}
```

The above code may also be valid.
Hence we need to think about `goto` statement is only available in the subroutine scope, and we should lint it correctly.

Reply:
Actually, you are totally right here! 
I missed this case even tho I have thought about it! 

Thanks for pointing it out, I will fix it